x/emissions/keeper/keeper.go

```diff
@@ -1607,28 +1607,38 @@ func (k Keeper) GetIdsOfActiveTopics(ctx context.Context, pagination *types.Simp
 	if err != nil {
 		return nil, nil, err
 	}
+	rng := new(collections.Range[uint64]).StartExclusive(start)
 
-	startKey := make([]byte, binary.MaxVarintLen64)
-	binary.BigEndian.PutUint64(startKey, start)
-	nextKey := make([]byte, binary.MaxVarintLen64)
-	binary.BigEndian.PutUint64(nextKey, start+limit)
-
-	rng, err := k.activeTopics.IterateRaw(ctx, startKey, nextKey, collections.OrderAscending)
+	iter, err := k.activeTopics.Iterate(ctx, rng)
 	if err != nil {
 		return nil, nil, err
 	}
-	activeTopics, err := rng.Keys()
-	if err != nil {
-		return nil, nil, err
+	defer iter.Close()
+
+	activeTopicIds := make([]TopicId, 0)
+	nextKey := make([]byte, binary.MaxVarintLen64)
+	nextTopicId := uint64(0)
+	for ; iter.Valid(); iter.Next() {
+		topicId, err := iter.Key()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if uint64(len(activeTopicIds)) >= limit {
```

Comment:
Hmm, is this how we want pagination to work? Shouldn't it iterate a limited number of times, regardless of what data is found? 

Otherwise someone could iterate much longer than the pagination limit if there are no active topics

This is an unlikely scenario, but it seems to possibly defeat the purpose of pagination

Reply:
In theory this has the same effect as keeping a counter that we increment at each iteration, and breaking once that reaches a threshold. Any way that logic is no good?

Reply:
well as you've written it the counter increments only if an activeTopicId is found, whereas in this comment it increments whether an active topic or inactive topic is found

Reply:
I don't have a strong opinion about this given that an attack is probably not viable

Reply:
ooopsss I misunderstood the code. We are only incrementing over active topics. Cool, this is correct.